agenthub/monologue_agent/utils/memory.py

```diff
@@ -84,7 +86,11 @@ def add_event(self, event: dict):
             },
         )
         self.thought_idx += 1
-        logger.debug("Adding %s event to memory: %d", t, self.thought_idx)
+        logger.debug('Adding %s event to memory: %d', t, self.thought_idx)
+        thread = Thread(target=self._add_doc, args=(doc,))
+        thread.start()  # We add the doc concurrently so we don't have to wait ~500ms for the insert
```

Comment:
I guess you missed thread.join() to wait for the execution until it finished?
```suggestion
        thread.start()  # We add the doc concurrently so we don't have to wait ~500ms for the insert
        thread.join()
```

Reply:
but LGMT! I approve

Reply:
Ah that's exactly what we _don't_ want to do! We want to kick off the `add_doc` in the background, and return immediately, so other work can proceed.